bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/FileChannelBackingCache.java

```diff
@@ -90,22 +88,18 @@ CachedFileChannel loadFileChannel(long logId) throws IOException {
      * @param cachedFileChannel
      */
     private void releaseFileChannel(long logId, CachedFileChannel cachedFileChannel) {
-        lock.writeLock().lock();
-        try {
-            if (cachedFileChannel.markDead()) {
-                try {
-                    cachedFileChannel.fileChannel.close();
-                } catch (IOException e) {
-                    LOG.warn("Exception occurred in ReferenceCountedFileChannel"
-                            + " while closing channel for log file: {}", cachedFileChannel);
-                } finally {
-                    IOUtils.close(LOG, cachedFileChannel.fileChannel);
-                }
-                // to guarantee the removed cachedFileChannel is what we want to remove.
-                fileChannels.remove(logId, cachedFileChannel);
+        if (cachedFileChannel.markDead()) {
+            // to guarantee the removed cachedFileChannel is what we want to remove.
+            fileChannels.remove(logId, cachedFileChannel);
+            // close corresponding fileChannel
+            try {
+                cachedFileChannel.fileChannel.close();
```

Comment:
it would be better is cachedFileChannel had a close method rather than directly accessing members.